catalog/plugin/catalog-plugin-metacardbackup-storage-common/src/test/java/org/codice/ddf/catalog/plugin/metacard/backup/common/MetacardStorageRouteTest.java

```diff
@@ -0,0 +1,251 @@
+                Boolean.class)).thenReturn(false);
+        Predicate predicate = storageRoute.getShouldBackupPredicate();
+        assertThat(predicate.matches(mockExchange), is(false));
+    }
+
+    @Test
+    public void testCheckDeletePredicateDontKeepBackups() {
+        setTestDefaults();
+        Exchange mockExchange = mock(Exchange.class);
+        Message mockMessage = mock(Message.class);
+        when(mockExchange.getIn()).thenReturn(mockMessage);
+        when(mockMessage.getHeader(PostIngestConsumer.ACTION, String.class)).thenReturn(
+                PostIngestConsumer.DELETE);
+        when(mockMessage.getHeader(MetacardStorageRoute.METACARD_BACKUP_KEEP_DELETED_RTE_PROP,
+                Boolean.class)).thenReturn(false);
+        Predicate predicate = storageRoute.getCheckDeletePredicate();
+        assertThat(predicate.matches(mockExchange), is(true));
+    }
+
+    @Test
+    public void testCheckDeletePredicateKeepBackups() {
+        setTestDefaults();
+        Exchange mockExchange = mock(Exchange.class);
+        Message mockMessage = mock(Message.class);
+        when(mockExchange.getIn()).thenReturn(mockMessage);
+        when(mockMessage.getHeader(PostIngestConsumer.ACTION, String.class)).thenReturn(
+                PostIngestConsumer.DELETE);
+        when(mockMessage.getHeader(MetacardStorageRoute.METACARD_BACKUP_KEEP_DELETED_RTE_PROP,
+                Boolean.class)).thenReturn(true);
+        Predicate predicate = storageRoute.getCheckDeletePredicate();
+        assertThat(predicate.matches(mockExchange), is(false));
+    }
+
+    @Test
+    public void testCheckDeletePredicateWhenCreateAction() {
+        setTestDefaults();
+        Exchange mockExchange = mock(Exchange.class);
+        Message mockMessage = mock(Message.class);
+        when(mockExchange.getIn()).thenReturn(mockMessage);
+        when(mockMessage.getHeader(PostIngestConsumer.ACTION, String.class)).thenReturn(
+                PostIngestConsumer.CREATE);
+        when(mockMessage.getHeader(MetacardStorageRoute.METACARD_BACKUP_KEEP_DELETED_RTE_PROP,
+                Boolean.class)).thenReturn(true);
+        Predicate predicate = storageRoute.getCheckDeletePredicate();
+        assertThat(predicate.matches(mockExchange), is(true));
+    }
+
+    @Test
+    public void testProperties() {
+        storageRoute.setKeepDeletedMetacards(false);
+        assertThat(storageRoute.isKeepDeletedMetacards(), is(false));
+
+        storageRoute.setBackupInvalidMetacards(false);
+        assertThat(storageRoute.isBackupInvalidMetacards(), is(false));
+
+        storageRoute.setMetacardTransformerId("test");
+        assertThat(storageRoute.getMetacardTransformerId(), is("test"));
+    }
+
+    @Test
+    public void testBackupTrueTagInvalid() {
+        boolean shouldBackup = MetacardStorageRoute.shouldBackupMetacard(getTestInvalidMetacard(), true);
+        MatcherAssert.assertThat(shouldBackup, Matchers.is(true));
```

Comment:
❓  I may have missed it, but why do we have 2 different `assertThat` methods being used?